ADPhantomApp/src/ADPhantom.cpp

```diff
@@ -1451,6 +1477,128 @@ void ADPhantom::phantomCameraTask()
+  int start_frame = 0;
+  int end_frame = 0;
+  int start_cine = 0;
+  int end_cine = 0;
+  int num_cines = 0;
+  int uni_frame_lim = false; //Whether frame limits are applied to all cines
+  bool rangeValid = true;
+  std::string response;
+  this->lock();
+  
+
+  while (1){
+    debug(functionName, "Waiting for the download command");
+    this->unlock();
+    status = epicsEventWait(this->startDownloadEventId_);
+    debug(functionName, "Download command recieved");
+    this->lock();
+    getIntegerParam(PHANTOM_LivePreview_, &preview);
+    if (preview){
+      setStringParam(ADStatusMessage, "Cannot download while live previewing");
+      setIntegerParam(ADStatus, ADStatusError);
+    }
+    else{
+
+      setStringParam(ADStatusMessage, "Downloading");
+      rangeValid = true;
+
+      //Read in total number of cines available
+      getIntegerParam(PHANTOM_GetCineCount_, &num_cines);
+
+      // Read in the number of frames to download
+      getIntegerParam(PHANTOM_DownloadStartFrame_, &start_frame);
+      getIntegerParam(PHANTOM_DownloadEndFrame_, &end_frame);
+      getIntegerParam(PHANTOM_DownloadStartCine_, &start_cine);
+      getIntegerParam(PHANTOM_DownloadEndCine_, &end_cine);
+      getIntegerParam(PHANTOM_DownloadFrameMode_, &uni_frame_lim);
+
+      debug(functionName, "Download start cine", start_cine);
+      debug(functionName, "Download end cine", end_cine);
+      debug(functionName, "Download start frame", start_frame);
+      debug(functionName, "Download end frame", end_frame);
+
+      if (start_cine < 1 || start_cine > num_cines){
+        rangeValid=false;
+        setStringParam(ADStatusMessage, "start_cine value invalid");
+      }  else if (end_cine < 1 || end_cine > num_cines){
+        rangeValid=false;
+        setStringParam(ADStatusMessage, "end_cine value invalid");
+      } else if(uni_frame_lim){ //Start frame and end frame are applied to every cine
+        int first_frame = 0;
+        int last_frame = 0;
+        for(int cine{start_cine}; cine <= end_cine; cine++){
+          debug(functionName, "Starting sanity checks on cine ", cine);
+          getIntegerParam(PHANTOM_CnFirstFrame_[cine], &first_frame);
+          getIntegerParam(PHANTOM_CnLastFrame_[cine], &last_frame);
+          if(start_frame < first_frame || start_frame > last_frame){
+            rangeValid=false;
+            char message[256];
+            sprintf(message, "start_frame value invalid in cine %d", cine);
+            setStringParam(ADStatusMessage, message);
+            break;
+          } else if(end_frame < first_frame || end_frame > last_frame ){
+            rangeValid=false;
+            char message[256];
+            sprintf(message, "end_frame value invalid in cine %d", cine);
+            setStringParam(ADStatusMessage, message);
+            break;
+          } else if(end_frame < start_frame){
+            rangeValid=false;
+            setStringParam(ADStatusMessage, "start_frame cannot be after end_frame");
```

Comment:
For any failures where you set the ADStatusMessage, it might be worth also logging an error message to the output, because if lots of things are going wrong the ADStatusMessage can quickly become overwritten with other messages.  If each error is logged it provides an opportunity for an engineer to look back over the logs and see the error history.